Backend/FcTypes.hs

```diff
@@ -38,27 +38,36 @@ isFcArrowTy (FcTyApp (FcTyApp (FcTyCon tc) ty1) ty2)
   | tc == fcArrowTyCon  = Just (ty1, ty2)
 isFcArrowTy _other_type = Nothing
 
+-- * Unit Type Constructor
+-- ----------------------------------------------------------------------------
+
+mkFcUnitTy :: FcType
+mkFcUnitTy = FcTyCon fcUnitTyCon
```

Comment:
if there is no name collision, I'd call this `fcUnitTy` instead (shorter and nicer):
```suggestion
fcUnitTy :: FcType
fcUnitTy = FcTyCon fcUnitTyCon
```